### DISCUSSION TITLE
What is the purpose of the possibility to write my own import when data is always written to a CSV file

### DISCUSSION THREAD

**Discussion by rabol** [Lukasss93/laravel-larex#43] [General]
I think this a great package, and was very happy to see that one could write a importer/exporter, but at the end of the day, it will be written to a csv file in the larex:import command.

How about allowing one to create a Writer as well, should be a 'easy' modification :)

**Comment by Lukasss93** [comment]
This is a great idea! Can you provide use cases for which this is necessary? 
Is it possible, instead, to put parameters in the config to pass to the csv reader/writer?

**Comment by rabol** [comment]
Hi

I for one would like to have the raw strings in a database so that I can make comparison, and a ui to modify the translations maybe even send to a api for translation.

Using csv, do have some disadvantages e.g. the delimiter is different depending of your locale.

The fact that there is a possibility to create a importer, implies that one can store the strings in somethings else than csv

**Comment by Lukasss93** [comment]
The core concept of this library is, as stated in the description:
> Laravel Larex lets you translate your whole Laravel application with a single CSV file.

An importer allows you to import strings from a source into the CSV file.
An exporter allows you to export strings from the CSV file to a target.

```mermaid
flowchart LR
 subgraph subGraph0["Importers"]
    direction TB
        i0[/"laravel"/]
        i1[/"json:group"/]
        i2[/"json:lang"/]
  end
 subgraph Exporters["Exporters"]
    direction TB
        e0[/"laravel"/]
        e1[/"json:group"/]
        e2[/"json:lang"/]
  end
    i0 & i1 & i2  -- import to --> CSV["CSVFile"]
    CSV -- export to --> e0 & e1 & e2
    style CSV stroke-width:2px,stroke-dasharray: 0
    style subGraph0 fill:none,stroke:#00C853
    style Exporters fill:none,stroke:#D50000
```

Your idea is very interesting, but it would require a radical transformation that would completely change the original function of the library. It would be simpler to develop this idea as a new project, although I'm not sure if something similar already exists or if it's worth the effort.

**Comment by rabol** [comment]
Sure I understand that the project is about writing to a csv file, no problem.

I have made my own fork and have implemented the possibility to configure a Writer and Reader which means that you can still use it as it is, but simply allow it to be extended to e.g. Database storage.
